pilot/pkg/model/push_context.go

```diff
@@ -985,6 +1002,32 @@ func (ps *PushContext) initDestinationRules(env *Environment) error {
 	return nil
 }
 
+// AuthPolicyForProxy returns the matching auth policy for a given proxy
+func (ps *PushContext) AuthPolicyForProxy(service *Service, port int, proxy *Proxy) *authn.Policy {
+	policy := ps.AuthPolicies[service.Hostname][port]
+	authPolicy := policy.Spec.(*authn.Policy)
+	var workloadPolicy *Config
+
+	if len(authPolicy.Targets) > 0 {
```

Comment:
Related to the previous question about what are `service` and `proxy` in the input, why do we need to do this match here?

Reply:
Also, we haven't support label-selector in authn policy anyway. And if we do, I think the more correct way to get the policy is changing the AuthPolicies map instead.

Reply:
@yangminzhu you asked this in my original PR [here](https://github.com/istio/istio/pull/16296#discussion_r316443340) but lets move the discussion regarding labels to this PR.
@diemtvu I'm not quite sure what `match` you are referring to - do you mean the check to make sure there is a matching target label with proxy.label? I was following the same logic from here: https://github.com/istio/istio/blob/master/pilot/pkg/model/config.go#L549

Reply:
I think I understand what you were getting at - the configstore lookup expects labels provided to be for a service instance, not for proxy instance.  Changes pushed

Reply:
This looks better, but may be not necessary for this transient time. Long story short, I don't think we need to consider labels yet, simply get the policy for host+port is enough.

To give some background for that suggestion:
- We've planned to change authN policy to use label-selector (aka workload selector). However, the design for authN policy v2 is still up in the air, so this is not yet done.
- Also, when it happen, service-selector should be deprecated. In other words, we either select by service name (host) or label, but not both.
- That's why I think the whole code will need to rewrite for that, including the AuthPolicies map, and may need to iterate all service instances. Labels should be extracted from service instances, not from function argument.
- For now, it should be enough to simply return the entry of the AuthPolicies if found. More or less, it's the same as calling `env.IstioConfigStore.AuthenticationPolicyForWorkload(svc, nil, port)`, except you cache this.

Reply:
Could you replace the map with a struct? You could put the map inside the struct as a field. This makes it easier if we later want to change the data structure here.